### ISSUE TITLE
Wrappers around ezf for more fzf functionality

### ISSUE THREAD

**Issue by atanasj** [mickeynp/ezf#2]
Great idea with this package. But how do I get it to work with `consult`, etc. I can get the candidates, but it does not insert it into the `shell-mode` buffer.

**Comment by atanasj** [comment]
Okay, I think I got it working  now. I had to change the flag from 1 to 8 to extract the name of the file... Also, I was not using the `$(args | ezf -f 8)` correctly in the call. 

I  know it's very early in the life of `ezf`, but it would be great to have a wrapper around some of its functionality to work more like `fzf`… i.e., instead of having to wrap what you want inside the `$(args | ezf -f 1)`... 

So, this is more a feature request than an issue.... Title changed accordingly.

**Comment by mickeynp** [comment]
You don't have to wrap things inside command substitution:

```
$ apt-cache search python | ezf -f 1 | cat
python3-cg3 prov-tools lola acr
```

You can pipe it to other things just fine.

**Comment by atanasj** [comment]
Thanks for that. I tried your suggestion, but that doesn't seem to work… Maybe I'm doing something wrong. See below. My expectation is that the `ezf` director will be added to the preceding `cd ~/git/`, but that does not happen.

```sh
~ ⭘ 
➤ cd ~/git/ | ezf -f 8 | cat
ezf/

~ ⭘ 
➤ 
```

However, when I do this: 

```sh
~ ⭘ 
➤ cd ~/git/$(cd ~/git/ | ezf -f 8)
total 364K
drwxr-xr-x 12 atanas  384 Jul 29 09:27 .git/
-rw-r--r--  1 atanas  35K Jul 29 08:15 LICENSE
-rw-r--r--  1 atanas 1.1K Jul 29 08:15 README.rst
-rwxr-xr-x  1 atanas  749 Jul 29 08:15 ezf
-rw-r--r--  1 atanas 314K Jul 29 08:15 ezf-example-optimised.gif
-rw-r--r--  1 atanas 3.0K Jul 29 08:57 ezf.el

~/git/ezf ⭘ 
➤ 
```
I get the expected behaviour... Am I dong something wrong?

**Comment by mickeynp** [comment]
I don't think that command would ever work. `cd` returns no output. How can you filter on it?

```
$ pwd
/
$ cd /$(echo "home")
$ pwd
/home
```

... but this works:

```
$ cd $(ls | ezf)
...
```

**Comment by atanasj** [comment]
Yes, that works, but maybe my understanding of how the functions work or the intentions are wrong. That is, the whatever candidate I select in the minibuffer, will get inserted where I want it at the prompt... In your article, in particular using `helm` with multi select, there seems to be lots of things you could possibly do and a function like `ezf` could be used for. 

The way I use `fzf` is to quickly jump to directories or find files, but there's probably other ways to that within `Emacs` which your post on shells and terminals with `Emacs` covers pretty well.

My suggestion of a wrapper (if that is even the right solution to what I need) was to make it more similar to what I know about `fzf`... I wonder if there is a way to do this, or maybe it's better if I think about other ways those problems (finding files, directories, etc.) can be handled / solved within `Emacs`.

**Comment by mickeynp** [comment]
I don't quite follow. If I swap ezf with fzf in both mine and your examples above, I get the exact same outcome.

I do not understand what it is you want to achieve? Can you please describe the exact workflow with fzf?
